server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -83,6 +84,7 @@ public void onCommit(List<? extends IndexCommit> commits) throws IOException {
             this.safeCommitInfo = SafeCommitInfo.EMPTY;
             this.lastCommit = commits.get(commits.size() - 1);
             this.safeCommit = commits.get(keptPosition);
+            this.nextSafeCommit = keptPosition == commits.size() - 1 ? null : commits.get(keptPosition + 1);
```

Comment:
`indexOfKeptCommits`filters out commits that dont match translog uuid. do we need to do the same here?

Reply:
Both the safe commit and the next safe commit should have the same translog UUID since we iterate the commit list in reverse.